### PR TITLE
Client URL scheme validation #6547

### LINKED ISSUE #6547: Nima - HTTP/2 Client - validate protocol

### PR DESCRIPTION
**Pull request by danielkec**
Fixes #6547

### CODE REVIEW

**Review by tomas-langer — CHANGES_REQUESTED** [review]
As per our direct messages:
The `ClientUri` represents a URI (that should be used regardless of protocol.
The validation is a function of the client request in use - so WebSocket client should validate ws, wss, http, https; Http based web clients should validate `http, https` etc.

File: webclient/websocket/src/test/java/io/helidon/webclient/websocket/WsClientTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.startsWith;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class WsClientTest {
+    @Test
+    void testSchemeValidation() {
+        try {
```
Comment (tomas-langer) [inline webclient/websocket/src/test/java/io/helidon/webclient/websocket/WsClientTest.java:31]: Use `assertThrows` instead

File: webclient/websocket/src/test/java/io/helidon/webclient/websocket/WsClientTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import io.helidon.websocket.WsListener;
+import io.helidon.websocket.WsSession;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.startsWith;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class WsClientTest {
```
Comment (tomas-langer) [inline webclient/websocket/src/test/java/io/helidon/webclient/websocket/WsClientTest.java:28]: Do not use `public`

File: webclient/websocket/src/test/java/io/helidon/webclient/websocket/WsClientTest.java (hunk 1 above)
Comment (danielkec) [reply webclient/websocket/src/test/java/io/helidon/webclient/websocket/WsClientTest.java]: fixed

File: webclient/websocket/src/test/java/io/helidon/webclient/websocket/WsClientTest.java (hunk 2 above)
Comment (danielkec) [reply webclient/websocket/src/test/java/io/helidon/webclient/websocket/WsClientTest.java]: fixed

**Review by tomas-langer — APPROVED** [review]

**Pull request merged**